lib/bemhtml/index.js

```diff
@@ -213,7 +213,8 @@ BEMHTML.prototype.renderMix = function renderMix(entity,
     if (typeof item === 'string')
       item = { block: item, elem: undefined };
 
-    var hasItem = item.block || item.elem;
+    var hasItem = (item.block || item.elem) &&
+      !(item.mods && item.block === entity.block);
```

Comment:
Pretty scared about this condition. Can we add more tests for edge cases?

Like:
```js
{ block: 'b', elem: 'e', mix: { elem: 'e' } } // 1
{ block: 'b', elem: 'e', mix: { block: 'b', elem: 'e' } } // 2
{ block: 'b', content: { elem: 'e', mix: { elem: 'e' } } } // 3
```